### ISSUE TITLE
Request - Add muscle tendon velocity as file output in MuscleAnalysis

### ISSUE THREAD

**Issue by CVHammond** [opensim-org/opensim-core#3172]
RCNL's NMSM Pipeline uses muscle tendon velocity as part of our Muscle Tendon Personalization (MTP) module which we hope will be part of the OpenSim GUI Tools within the next year.

We are hoping that GUI-only users will be able to generate all of the data needed by MTP from within the GUI from the `IK`, `ID`, and `MuscleAnalysis` tools. After speaking with @aymanhab, we are requesting the addition of a muscle tendon velocity file as output from `MuscleAnalysis` in the GUI.

We are under the impression that the implementation of this addition is relatively simple given that fiber velocity is already calculated as part of `MuscleAnalysis`.

As far as I can see in the documentation: a `getMuscleTendonVelocityStorage()` function needs to be added to the `OpenSim::MuscleAnalysis` class as well as the underlying logic to record to a new muscle tendon velocity `Storage` object during the `MuscleAnalysis` frame stepping function. After this, the muscle tendon velocity needs to be added as an option within the GUI, or, more easily, printing muscle tendon velocity to a file can be added to the default result files of `MuscleAnalysis`.

Please let me know if you need additional information.

**Comment by carmichaelong** [comment]
@CVHammond Some documentation on the Output Reporter: https://simtk-confluence.stanford.edu:8443/display/OpenSim/Example+-+Output+Reporter+with+a+Jumping+Simulation

@aymanhab I haven't tendon velocity come up too much, but I'm realizing that if you have a lot of muscles and you want the same output for all of them, this might be onerous for GUI users if the output paths can't use wildcard, regex, etc (I skimmed the code and it doesn't look like it can, but I'm not 100% sure). In that case for ease of use, it might be helpful to add to `MuscleAnalysis` and/or more flexible ways to pass in groups of outputs.

**Comment by CVHammond** [comment]
@aymanhab When using the `OutputReporter` if I attempt to request a muscle speed (i.e. `forceset/addbrev_r|speed`) the following error occurs:

`SimTK Exception thrown at State.cpp:974: Error detected by Simbody method getCacheEntry: State Cache entry was out of date at Stage Velocity. This entry depends on version 2 of Stage Velocity but was last updated at version 0.`

**Comment by aymanhab** [comment]
@CVHammond Some context maybe helpful but generally you get this error when you're trying to access cache entry that depends on Velocity stage while the state has not been realized to Velocity. Try calling realizeVelocity first or find out what change was made to the model causing the realization to go down to Topology

**Comment by CVHammond** [comment]
Sorry, I discussed the definition of MuscleTendonVelocity with our team and there was confusion regarding how it was calculated. This feature request was based on the idea that MuscleTendonVelocity was calculated inherently during MuscleAnalysis and just needed to be outputted. After looking at the implementation more closely, we have decided to calculate MuscleTendonVelocity after the fact using a technique within our module. Thank you for your help regardless
